persistence/pre_persistence/account.go

```diff
@@ -0,0 +1,377 @@
+package pre_persistence
+
+import (
+	"bytes"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/types"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"math/big"
+)
+
+const (
+	ServiceNodeStakePoolName = "SERVICE_NODE_STAKE_POOL"
+	AppStakePoolName         = "APP_STAKE_POOL"
+	ValidatorStakePoolName   = "VALIDATOR_STAKE_POOL"
+	FishermanStakePoolName   = "FISHERMAN_STAKE_POOL"
+	DAOPoolName              = "DAO_POOL"
+	FeePoolName              = "FEE_POOL"
+)
+
+func (m *PrePersistenceContext) AddPoolAmount(name string, amount string) error {
+	cdc := Cdc()
+	p := Pool{}
+	db := m.Store()
+	key := append(PoolPrefixKey, []byte(name)...)
+	val, err := db.Get(key)
+	if err != nil {
+		return err
+	}
+	err = cdc.Unmarshal(val, &p)
+	if err != nil {
+		return err
+	}
+	s, err := StringToBigInt(p.Account.Amount)
+	if err != nil {
+		return err
+	}
+	s2, err := StringToBigInt(amount)
+	if err != nil {
+		return err
+	}
+	s.Add(s, s2)
+	p.Account.Amount = BigIntToString(s)
+	bz, err := cdc.Marshal(&p)
+	if err != nil {
+		return err
+	}
+	return db.Put(key, bz)
+}
+
+func (m *PrePersistenceContext) SubtractPoolAmount(name string, amount string) error {
+	cdc := Cdc()
+	p := Pool{}
+	db := m.Store()
+	key := append(PoolPrefixKey, []byte(name)...)
+	val, err := db.Get(key)
+	if err != nil {
+		return err
+	}
+	err = cdc.Unmarshal(val, &p)
+	if err != nil {
+		return err
+	}
+	s, err := StringToBigInt(p.Account.Amount)
+	if err != nil {
+		return err
+	}
+	s2, err := StringToBigInt(amount)
+	if err != nil {
+		return err
+	}
+	s.Sub(s, s2)
+	p.Account.Amount = BigIntToString(s)
+	bz, err := cdc.Marshal(&p)
+	if err != nil {
+		return err
+	}
+	return db.Put(key, bz)
+}
+
+func (m *PrePersistenceContext) InsertPool(name string, address []byte, amount string) error {
+	cdc := Cdc()
+	p := Pool{
+		Name: name,
+		Account: &Account{
+			Address: address,
+			Amount:  amount,
+		},
+	}
+	db := m.Store()
+	key := append(PoolPrefixKey, []byte(name)...)
+	bz, err := cdc.Marshal(&p)
+	if err != nil {
+		return err
+	}
+	return db.Put(key, bz)
+}
+
+func (m *PrePersistenceContext) SetPoolAmount(name string, amount string) error {
+	cdc := Cdc()
```

Comment:
To discuss: per my comment above, I think it is pretty easy to generalize out a lot of the boilerplate in every function into a helper.

Reply:
Maybe we can work together to get there. The comment is a bit too generic for me to take action

Reply:
I left a code block example in one of my previous comments: https://github.com/pokt-network/pocket/pull/42#discussion_r818120191

Reply:
Yes, and I used that code where commented for the add and subtract functions.

But specifically for the setter, what kind of 'boilerplate' are you thinking for 'every function'? It's a bit too generic for met to take any action

Reply:
After reviewing utility, I think the verbosity is okay, especially since it's pre-persistence.

We can generalize things over time (in utility) as we see fit.